configure.ac

```diff
@@ -115,7 +115,7 @@ AC_SUBST([HARDEN_CFLAGS])
 AC_SUBST([HARDEN_LDFLAGS])
 
 # Also check for a few non-hardening-related flags.
-MISC_CXXFLAGS=""
+MISC_CXXFLAGS="-std=c++11"
```

Comment:
If it were 2017 already, I'd be comfortable with using C++11 :)

Right now, Ubuntu 12.04 and RHEL 5 are still supported releases, and neither have C++11 in their default compiler.  But most of C++11 was stable and available well before that, so it's a difficult call.  FreeBSD i386/amd64/arm/arm64 have moved to clang, OpenBSD is still using gcc 4.2, non-mainstream architectures tend to lag somewhat in their compiler support.  I lean towards saying "Not this year".  Then again, the next release might well be next year anyway.

It's not just compilers, a C++11 standard library and headers is also needed.  Another issue is that we'd need to get autoconf C++ detection/configuration working reliably across a variety of systems.